### PR TITLE
Enhance solution formatting and cleanup post Rider 2023.2.3 upgrade

### PR DESCRIPTION
**Pull request by tjementum**
### Summary & Motivation

Perform a cleanup across the entire solution following the upgrade to Rider 2023.2.3. Move JetBrains formatting rules from `.editorconfig` to a .DotSettings file. This change enforces JetBrains formatting out-of-the-box and offers more granular control.

Add a `.NET only` cleanup profile to .DotSettings to prevent ReSharper and Rider from formatting frontend files. To maintain consistent formatting in frontend files, "Format on Save" should be manually configured to use Prettier. Additionally, file types such as `.html`, `.css`, and `.json` need to be added to Prettier's configuration for comprehensive coverage.

Furthermore, .NET tools, including the JetBrains formatter, have been added to `dotnet-tools.json`. This inclusion streamlines tool management and ensures consistency across development environments. The GitHub workflow has been correspondingly updated to reflect these tooling changes, enhancing the efficiency and reliability of our continuous integration process.

### Checklist

- [x] I have added a Label to the pull-request
- [x] I have added tests, and done manual regression tests
- [x] I have updated the documentation, if necessary

### CODE REVIEW

File: application/PlatformPlatform.sln.DotSettings [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  &amp;lt;option name="myName" value=".NET only" /&amp;gt;
```
Comment (raix) [inline application/PlatformPlatform.sln.DotSettings:3]: "myName" seems a bit random

**Review by raix — COMMENTED** [review]
Would this config also work in Visual Studio?

**Comment by tjementum** [comment]
> Would this config also work in Visual Studio?



> Would this config also work in Visual Studio?

Yes... This is actually a ReSharper file. My experience is it works exactly the same.

File: application/PlatformPlatform.sln.DotSettings (hunk 1 above)
Comment (tjementum) [reply application/PlatformPlatform.sln.DotSettings:3]: It's generated by Rider. But I've changed it to `profileName`.

File: application/PlatformPlatform.sln.DotSettings [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  &amp;lt;inspection_tool class="JSRemoveUnnecessaryParentheses" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="JSUnnecessarySemicolon" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="TypeScriptExplicitMemberType" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="UnnecessaryContinueJS" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="UnnecessaryLabelJS" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="UnnecessaryLabelOnBreakStatementJS" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="UnnecessaryLabelOnContinueStatementJS" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="UnnecessaryReturnJS" enabled="false" level="WARNING" enabled_by_default="false" /&amp;gt;
+  &amp;lt;inspection_tool class="WrongPropertyKeyValueDelimiter" enabled="false" level="WEAK WARNING" enabled_by_default="false" /&amp;gt;
```
[the remark this answers is not in the document]
Comment (tjementum) [reply application/PlatformPlatform.sln.DotSettings:15]: This is for ensuring that Rider does not "complain" about frontend stuff. Not sure if Rider will accept ESLint... but let's see.

**Comment by tjementum** [comment]
Ups... I clicked close pull-request instead of Resolve comment. I will recreate.

**Comment by tjementum** [comment]
Ups... I clicked close pull-request instead of Resolve comment. I will reopen.

**Pull request closed without merging**